cirq-core/cirq/work/observable_measurement_data.py

```diff
@@ -132,6 +134,17 @@ def observable(self):
     def stddev(self):
         return np.sqrt(self.variance)
 
+    def as_dict(self) -> Dict[str, Any]:
+        """Return the contents of this class as a dictionary.
+
+        This makes records suitable for construction of a Pandas dataframe. The circuit parameters
+        are flattened into the top-level of this dictionary.
+        """
+        record = dataclasses.asdict(self)
+        del record['circuit_params']
+        record.update(**self.circuit_params)
```

Comment:
Do you think we should check for collisions between `record` and `self.circuit_params` here? It seems unlikely that someone would use overlapping parameter names, but if they did it would be really tricky to spot what went wrong.

Reply:
You're right. What should the behavior be in this case? an error? that could be pretty annoying especially if it's after a long data collection run. Prefix the new key names with `"circuit_params."`? Probably. Should we only do the prefixing if a collision would have happened, or always?

Reply:
prefixed names with "circuit_params." if and only if *any* of the circuit params would collide

Reply:
Agreed that we shouldn't return an error, though personally I'm in favor of consistency (i.e. always prefix).

Reply:
changed to always prefixing